db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2879,6 +2882,9 @@ void DBImpl::SchedulePendingFlush(const FlushRequest& flush_req) {
 
 void DBImpl::SchedulePendingCompaction(ColumnFamilyData* cfd) {
```

Comment:
There are three places in this file that are not using `DBImpl::SchedulePendingCompaction` to schedule compactions but instead using the code below:

```
AddToCompactionQueue(cfd);
++unscheduled_compactions_;
```

These need to be refactored to call this function instead so the guarding flag can guard them too.

Reply:
Let's discuss this offline. If we are addressing this, I'd like to do this in a separate PR.